wys/api/sql/function-aggregate_speed_counts_one_hour_5kph.sql

```diff
@@ -16,34 +16,38 @@ BEGIN
 
 	WITH insert_data AS (
 		--Aggregated into speed bins and 1 hour bin
-		INSERT INTO wys.speed_counts_agg_5kph (api_id, datetime_bin, speed_id, volume)
-		SELECT api_id, date_trunc('hour', datetime_bin) dt, speed_id, sum(count) AS volume
-		FROM wys.raw_data
-		INNER JOIN wys.speed_bins_old ON speed <@ speed_bin
-		WHERE	speed_count_uid IS NULL
-		GROUP BY api_id, dt,  speed_id 
-		RETURNING speed_counts_agg_5kph_id, api_id, datetime_bin, speed_id
+		INSERT INTO 	wys.speed_counts_agg_5kph (api_id, datetime_bin, speed_id, volume)
+		SELECT 			api_id, date_trunc('hour', datetime_bin) hr, speed_id, sum(count) AS volume
+		FROM 			wys.raw_data
+		INNER JOIN 		wys.speed_bins_old ON speed <@ speed_bin
+		WHERE			speed_count_uid IS NULL AND 
+                        "count" IS NOT NULL AND 
+                        datetime_bin >= _start_date AND datetime_bin < _end_date
+        
+		GROUP BY 		api_id, hr,  speed_id 
+		RETURNING 		speed_counts_agg_5kph_id, api_id, datetime_bin, speed_id
 		)
 	
-	UPDATE wys.raw_data A
-	SET speed_count_uid=B.speed_counts_agg_5kph_id
-	FROM insert_data B
-	WHERE A.speed_count_uid IS NULL
-	AND A.api_id=B.api_id
-	AND A.datetime_bin >= B.datetime_bin AND A.datetime_bin < B.datetime_bin + INTERVAL '1 hour';
+	UPDATE 	wys.raw_data A
+	SET 	speed_count_uid=B.speed_counts_agg_5kph_id
+	FROM 	insert_data B
+	WHERE 	A.speed_count_uid IS NULL AND
```

Comment:
Might speed things up to add `datetime_bin >= _start_date AND datetime_bin < _end_date` here